packages/material-ui/src/styles/ThemeProvider.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles';
+import { exactProp } from '@material-ui/utils';
+import { ThemeContext as StyledEngineThemeContext } from '@material-ui/styled-engine';
+import useTheme from './useTheme';
+
+function InnerThemeProvider({ children }) {
+  const theme = useTheme();
+  return (
+    <StyledEngineThemeContext.Provider value={typeof theme === 'object' ? theme : {}}>
+      {children}
+    </StyledEngineThemeContext.Provider>
+  );
+}
+
+InnerThemeProvider.propTypes = {
+  /**
+   * Your component tree.
+   */
+  children: PropTypes.node,
+};
+
+/**
+ * This component takes a `theme` prop.
+ * It makes the `theme` available down the React tree thanks to React context.
```

Comment:
suggestion: 
```suggestion
 * It makes the `theme` available down the React tree.
```

React context is an implementation detail. What props it takes is available in the props API.